src/lib/import_export/csv_parser.cpp

```diff
@@ -18,10 +19,19 @@
 
 namespace opossum {
 
-CsvParser::CsvParser(const CsvConfig& csv_config) : _csv_config(csv_config) {}
+CsvParser::CsvParser() {}
 
-std::shared_ptr<Table> CsvParser::parse(const std::string& filename) {
-  const auto table = _process_meta_file(filename + _csv_config.meta_file_extension);
+std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta> csv_meta) {
+  if (csv_meta == std::nullopt) {
```

Comment:
This needs a comment. "If no meta info is given as a parameter, look for a json file".

Do we want to handle the case that a meta file exists but gets ignored?

Reply:
Added the comment. As of handling, I don't think it is needed. If someone explicitly passes a custom CsvMeta, then I'd assume they know what they are doing